kentik_api_library/kentik_api/analytics/mapped_query.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Callable, Dict, Generator, Optional, Tuple, Type, TypeVar
+
+import yaml
+from pandas import DataFrame, to_datetime
+
+from kentik_api import KentikAPI
+from kentik_api.public import QuerySQL, QuerySQLResult
+
+MappedQueryFn = Callable[..., DataFrame]
+
+
+@dataclass
+class MappingEntry:
+    format: str
+    data_type: Optional[str]
+    is_index: bool = False
+
+
+T = TypeVar("T", bound="SQLResultMapping")
+
+
+class SQLResultMapping:
+    @classmethod
+    def from_dict(cls: Type[T], data: Optional[Dict] = None) -> T:
```

Comment:
What is the difference in providing None as data vs providing empty dict as data?

Reply:
And in general, providing None as container instead of empty container?

Reply:
OK. answered in another place